### PR TITLE
implement a `split` filter

### PR DESCRIPTION
**Pull request by wolfv**
I think it's kinda useful, especially as a counterpart to `join`.

### CODE REVIEW

File: minijinja/src/filters.rs [hunk 1]
```diff
@@ -557,6 +557,23 @@ mod builtins {
         }
     }
 
+    /// Split a string with a separator
```
Comment (mitsuhiko) [inline minijinja/src/filters.rs:560]: Can you add an example here?

File: minijinja/src/filters.rs [hunk 2]
```diff
@@ -557,6 +557,23 @@ mod builtins {
...
+    #[cfg_attr(docsrs, doc(cfg(feature = "builtins")))]
+    pub fn split(val: Value, split: Option<Cow<'_, str>>) -> Result<Value, Error> {
```
Comment (mitsuhiko) [inline minijinja/src/filters.rs:562]: I would change this filter in two ways so it's similar to the `.split` function in Python:

- if `None` or an empty string is passed (or nothing at all), it splits on any whitespace (`split_whitespace`).
- a second parameter (`nsplits`) is added which indicates how many times the string is to be split. `1` means one split (eg: two parts), `2` means two splits (eg: three parts).

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

File: minijinja/src/filters.rs [hunk 3]
```diff
@@ -557,6 +557,23 @@ mod builtins {
...
+        let split = split.as_deref().unwrap_or(" ");
+
+        match val.as_str() {
+            Some(s) => {
+                let elements = s.split(split).map(Value::from).collect::<Vec<Value>>();
```
Comment (mitsuhiko) [inline minijinja/src/filters.rs:567]: This could be returned as iterable rather than collected into a Vec

File: minijinja/src/filters.rs (hunk 2 above)
Comment (wolfv) [reply minijinja/src/filters.rs:566]: Alright - yeah, I was looking for your input before continuing. Sounds like a plan!

**Pull request merged**